nixos/modules/services/hardware/actkbd.nix

```diff
@@ -6,18 +6,47 @@ let
 
   cfg = config.services.actkbd;
 
-  configFile = pkgs.writeText "actkbd.conf" ''
+  isUserBindings = bool:
+    builtins.filter (a: a.userMode == bool) cfg.bindings;
+
+  mkConfigFile = isUser:
+    pkgs.writeText "actkbd.conf" ''
     ${concatMapStringsSep "\n"
       ({ keys, events, attributes, command, ... }:
         ''${concatMapStringsSep "+" toString keys}:${concatStringsSep "," events}:${concatStringsSep "," attributes}:${command}''
       )
-      cfg.bindings}
-    ${cfg.extraConfig}
-  '';
+      ( isUserBindings isUser )}
+      ${cfg.extraConfig}'';
+
+  mkSystemdService = config:
+  {
+    enable = true;
+    restartIfChanged = true;
+    unitConfig = {
+      Description = "actkbd on %I";
+      ConditionPathExists = "%I";
+    };
+    serviceConfig = {
+      Type = "forking";
```

Comment:
I believe forking could be changed to simple and -D removed from the ExecStart -line.